source/common/network/dns_impl.h

```diff
@@ -50,17 +51,18 @@ class DnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logger::I
     }
 
     /**
-     * c-ares ares_gethostbyname() query callback.
-     * @param status return status of call to ares_gethostbyname.
+     * ares_getaddrinfo query callback.
+     * @param status return status of call to ares_getaddrinfo.
      * @param timeouts the number of times the request timed out.
-     * @param hostent structure that stores information about a given host.
+     * @param addrinfo structure to store address info.
      */
-    void onAresHostCallback(int status, int timeouts, hostent* hostent);
+    void onAresGetAddrInfoCallback(int status, int timeouts, ares_addrinfo* addrinfo);
```

Comment:
Not sure how hard it is, but ideally a PR like this could be split into (1) switch to `ares_getaddrinfo` and (2) TTL refresh rate. That would make it much simpler to tackle as a reviewer.

Reply:
I could split the pr into two. The first PR seems to be a small change to me so I merge them together.

Reply:
Yeah, I think that would simplify, thanks.